### PR TITLE
Add parameter for "License Type" when using an advanced image.

### LINKED ISSUE #506: AzureVMAgent template donot have attribute to set license_type
### What feature do you want to see added?

No attribute available to set license type as "windows_client/windows_server" for the agents launching with AzureSSHLauncher 

### Upstream changes

_No response_

### Are you interested in contributing this feature?

_No response_

### PR DESCRIPTION
**Pull request by krische**
This adds an additional "License Type" property to advanced image configuration. This allows the setting of of the `licenseType` property on Windows VMs to take advantage of the [Azure Hybrid Benefit](https://learn.microsoft.com/en-us/azure/virtual-machines/windows/hybrid-use-benefit-licensing) licensing feature.

Overall a rather simple change. I added a new selection dropdown for "License Type" beneath the "OS Type" property when editing the configuration of an advanced image. The default is `Classic` which means nothing will be done. Only when the value is `Windows_Client` or `Windows_Server` will the `licenseType` property be added to the deployment.

This should resolve #506.

### Testing done

Added to the JCasC tests to verify the new parameter can be imported and exported.

Here is a screenshot of what the dropdown looks like in the UI:
image

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java [hunk 1]
```diff
@@ -428,7 +430,8 @@ public AzureVMAgentTemplate(
             String jvmOptions,
             RetentionStrategy<?> retentionStrategy,
             boolean executeInitScriptAsRoot,
-            boolean doNotUseMachineIfInitFails
+            boolean doNotUseMachineIfInitFails,
+            String licenseType
```
Comment (timja) [inline src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java:434]: can you add this as a setter annotated with databoundsetter please so its not a breaking change for JCasC users. Then you can revert the changes to the yaml files and this will be good to go.

**Review by timja — CHANGES_REQUESTED** [review]

File: src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java (hunk 1 above)
Comment (krische) [reply src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java]: Okay, removed it from the constructor everywhere.

**Review by timja — APPROVED** [review]
I've not tested this locally (I don't have the hybrid benefits anyway) but it looks sensible and isolated.

Thanks for the pull request!
